### PR TITLE
 Remove password from user edit field, add validate password to user

### PR DESCRIPTION
**Pull request by apalski**
PR to remove password field from edit user
- Add validate_password? method to user
- Add if validate_password? to password length validation
- Remove password field from edit form
- Remove password field from edit form spec
All feedback and suggestions welcome!
https://trello.com/c/33Dh5bWw/
Thanks for taking a look!

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -5,9 +5,15 @@ class User < ActiveRecord::Base
 
   validates :name, :email, presence: true, uniqueness: true
   validates :email, confirmation: true
-  validates :password, length: { in: 6..64 }
+  validates :password, length: { in: 6..64, if: :validate_password? }
```
Comment (elle) [inline app/models/user.rb]: A suggestion to change this line to:
` validates :password, length: { in: 6..64 }, allow_nil: true`

File: app/models/user.rb [hunk 2]
```diff
@@ -5,9 +5,15 @@ class User < ActiveRecord::Base
...
 
   def self.by_admin
     order(admin: :desc, name: :asc)
   end
+
+  private
+
+  def validate_password?
+    password.present?
+  end
```
Comment (elle) [inline app/models/user.rb]: And if following the previous comment, then this addition is not needed :)

File: app/models/user.rb (hunk 1 above)
Comment (apalski) [reply app/models/user.rb]: Fantastic solution, thank you, done in sha : 27867eb

File: app/models/user.rb (hunk 2 above)
Comment (apalski) [reply app/models/user.rb]: done in sha : 27867eb

**Comment by apalski** [comment]
Merging now, thank you for your feedback @elle

**Pull request closed without merging**